### PR TITLE
Laravel boot hooking (using event system for observability)

### PR DESCRIPTION
**Pull request by pdelewski**
Shows how can we bring https://github.com/plunkettscott/laravel-otel work into otel auto-instrumentation.
- initial boot hooking
- added abstract Watcher class
- added ClientRequestWatcher
- added ExceptionWatcher
- added CacheWatcher
- added LogWatcher
- added QueryWatcher
- Laravel tests dependencies

### CODE REVIEW

**Review by bobstrecansky — COMMENTED** [review]
LGTM to me so far.  Any explicit feedback you're looking for?

File: src/Instrumentation/Laravel/src/LaravelInstrumentation.php [hunk 1]
```diff
@@ -16,10 +21,112 @@
...
+        $processedUrl = $parsedUrl->get('scheme') . '://' . $parsedUrl->get('host') . $parsedUrl->get('path', '');
+
+        if ($parsedUrl->has('query')) {
+            $processedUrl .= '?' . $parsedUrl->get('query');
+        }
+
+        $span = $this->instrumentation->tracer()->spanBuilder('http ' . $request->request->method() . ' ' . $request->request->url())
+            ->setSpanKind(SpanKind::KIND_CLIENT)
+            ->setAttributes([
+                'http.method' => $request->request->method(),
```
Comment (brettmc) [inline src/Instrumentation/Laravel/src/LaravelInstrumentation.php:51]: we have constants in `TraceAttributes` for these

**Comment by brettmc** [comment]
@pdelewski this looks really good to me!

**Comment by bobstrecansky** [comment]
Sounds good to me.  Thank you!

**Comment by pdelewski** [comment]
@brettmc @bobstrecansky It's ready for review

**Review by bobstrecansky — APPROVED** [review]

File: src/Instrumentation/Laravel/src/ExceptionWatcher.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        if (! isset($log->context['exception']) ||
+        ! $log->context['exception'] instanceof Throwable) {
+            return;
+        }
+
+        $exception = $log->context['exception'];
+
+        $attributes = [
+            'code.namespace' => get_class($exception),
```
Comment (brettmc) [inline src/Instrumentation/Laravel/src/ExceptionWatcher.php:35]: can use TraceAttributes for these keys

File: src/Instrumentation/Laravel/tests/app/Console/Kernel.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+use Illuminate\Foundation\Console\Kernel as ConsoleKernel;
+
+class Kernel extends ConsoleKernel
+{
+    /**
+     * Define the application's command schedule.
+     */
+    protected function schedule(Schedule $schedule): void
+    {
+        // $schedule->command('inspire')->hourly();
```
Comment (brettmc) [inline src/Instrumentation/Laravel/tests/app/Console/Kernel.php:17]: needed?

Comment (pdelewski) [reply src/Instrumentation/Laravel/tests/app/Console/Kernel.php:17]: That's not needed, but that is automatically generated code by `Laravel`, have not touch it

File: src/Instrumentation/Laravel/src/ExceptionWatcher.php (hunk 2 above)
Comment (pdelewski) [reply src/Instrumentation/Laravel/src/ExceptionWatcher.php]: done

**Pull request merged**